iothub/service/src/Common/Exceptions/ErrorCode.cs

```diff
@@ -23,47 +28,101 @@ public enum ErrorCode
         BulkRegistryOperationFailure = 400013,
         CannotRegisterModuleToModule = 400301,
 
-        //Unauthorized - 401
+        // Unauthorized - 401
         IotHubNotFound = 401001,
+
         IotHubUnauthorizedAccess = 401002,
+
+        /// <summary>
+        /// The SAS token has expired or IoT hub couldn't authenticate the authentication header, rule, or key.
+        /// </summary>
         IotHubUnauthorized = 401003,
 
         // Forbidden - 403
         IotHubSuspended = 403001,
+
+        /// <summary>
+        /// The daily message quota for the IoT hub is exceeded.
+        /// </summary>
         IotHubQuotaExceeded = 403002,
+
         JobQuotaExceeded = 403003,
+
+        /// <summary>
+        /// The underlying cause is that the number of messages enqueued for the device exceeds the queue limit (50).
+        /// The most likely reason that you're running into this limit is because you're using HTTPS to receive the message,
+        /// which leads to continuous polling using ReceiveAsync, resulting in IoT hub throttling the request.
+        /// </summary>
         DeviceMaximumQueueDepthExceeded = 403004,
+
         IotHubMaxCbsTokenExceeded = 403005,
 
         // NotFound - 404
+
+        /// <summary>
+        /// The operation failed because the device cannot be found by IoT Hub. The device is either not registered or disabled.
+        /// </summary>
         DeviceNotFound = 404001,
+
         JobNotFound = 404002,
         PartitionNotFound = 404003,
         ModuleNotFound = 404010,
 
-        //Conflict - 409
+        // Conflict - 409
+
+        /// <summary>
+        /// There's already a device with the same device Id in the IoT hub.
+        /// </summary>
         DeviceAlreadyExists = 409001,
+
         ModuleAlreadyExistsOnDevice = 409301,
 
-        //PreconditionFailed - 412
+        // PreconditionFailed - 412
         PreconditionFailed = 412001,
+
+        /// <summary>
+        /// When a device receives a cloud-to-device message from the queue (for example, using ReceiveAsync())
+        /// the message is locked by IoT hub for a lock timeout duration of one minute.
+        /// If the device tries to complete the message after the lock timeout expires, IoT hub throws this exception.
+        /// </summary>
         DeviceMessageLockLost = 412002,
 
-        //RequestEntityTooLarge - 413
+        // RequestEntityTooLarge - 413
         MessageTooLarge = 413001,
+
         TooManyDevices = 413002,
         TooManyModulesOnDevice = 413003,
 
-        //Throttling Exception 
+        // Throttling Exception
```

Comment:
Nit: We can remove these one line comments now since the comments you added is lot more detailed :)

Reply:
These are just grouping the different exceptions. I don't have details for all comments so I just left the grouping there.

Reply:
Ok